backend/app/src/main/java/com/moh/phlat/backend/service/SourceDataServiceImpl.java

```diff
@@ -35,81 +36,106 @@ public class SourceDataServiceImpl implements SourceDataService {
 	private SourceDataFilterSpecification specificationService = new SourceDataFilterSpecificationImpl();
 	
 	@Override
-	public List<SourceData> getSourceData(Long controlId, SourceDataFilterParams filterProcess) {
-		
-		Specification<SourceData> combinedSpecification;
+	public List<SourceData> getSourceData(Long controlId, SourceDataFilterParams filterSource, Pageable pageable) {
 		
 		try {
-			combinedSpecification = specificationService.buildSpecificationWhereEqual("controlTableId", controlId.toString());
+			return sourceDataRepository.findAll(buildSpecification(controlId, filterSource), pageable);
 		} catch (Exception e) {
 			logger.error("Error occured: {}", e.getMessage(), e);
 			
 			return new ArrayList();
 		}
+	}
+	
+	@Override
+	public Long countSourceData(Long controlId,  SourceDataFilterParams filterSource) {
```

Comment:
```suggestion
	public long countSourceData(Long controlId,  SourceDataFilterParams filterParams) {
```
any reason for wrapper instead of primitive?

Reply:
0 is a valid number to be returned.  Null which isn't valid in a primitive is being sent back if there is an error.
Code was removed as not being necessary.

Reply:
isn't returning null fails the entire calculation and we should let the error be handled in our global exception handler?